core/domain/user_services_test.py

```diff
@@ -770,3 +770,129 @@ def utcnow(cls):
+            self.EXP_ID, self.owner_id, end_state_name='End')
+
+    def edit_exploration(self):
+        """Edit an existing exploration as EDITOR."""
+        self.login(self.EDITOR_EMAIL)
+        exp_services.update_exploration(self.editor_id, self.EXP_ID, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+        self.logout()
+
+    def reduce_last_edited_an_exploration(self, delta_hours):
+        user_settings = user_services.get_user_settings(self.editor_id)
+        user_settings.last_edited_an_exploration = (
+            user_settings.last_edited_an_exploration -
+            datetime.timedelta(hours=delta_hours))
+        user_services._save_user_settings(user_settings) # pylint: disable=protected-access
+
+    def test_legacy_user(self):
+        """Test the case of a user who are editing exploration for first time
+        after the last edited time check was introduced.
+        """
+        # Set up a 'previous-generation' user.
+        user_settings = user_services.get_user_settings(self.editor_id)
+        user_settings.last_edited_an_exploration = None
+        user_services._save_user_settings(user_settings)  # pylint: disable=protected-access
+
+        self.assertIsNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+        self.edit_exploration()
+
+        self.assertIsNotNone(
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+
+    def test_last_exp_edit_time_gets_updated(self):
+        self.edit_exploration()
+        self.reduce_last_edited_an_exploration(delta_hours=13)
+        previous_last_edited_an_exploration = (
+            user_services.get_user_settings(self.editor_id).
+            last_edited_an_exploration)
+        self.assertIsNotNone(previous_last_edited_an_exploration)
+
+        # Editing exploration after delta hours time.
+        self.edit_exploration()
+
+        self.assertGreater(
+            (user_services.get_user_settings(self.editor_id).
+             last_edited_an_exploration),
+            previous_last_edited_an_exploration)
+
+
+class LastExplorationCreatedIntegrationTests(test_utils.GenericTestBase):
+    EXP_ID_A = 'exp_a'
+    EXP_ID_B = 'exp_b'
+
+    def setUp(self):
+        """Create user for creating exploration."""
+        super(LastExplorationCreatedIntegrationTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+    def create_exploration(self, exp_id):
+        """Edit an existing exploration as EDITOR."""
+        self.login(self.OWNER_EMAIL)
+        self.save_new_valid_exploration(
+            exp_id, self.owner_id, end_state_name='End')
+        self.logout()
+
+    def reduce_last_created_an_exploration(self, delta_hours):
```

Comment:
That said, you only really use this once. I suggest just having it inline, and putting a comment above the block of code to explain what's going on since this is an unusual operation.